org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -3924,7 +3924,9 @@ private void completionOnSingleNameReference(ASTNode astNode, ASTNode astNodePar
 	}
 
 	private void checkForVarargExpectedTypes(ASTNode astNodeParent, Scope scope) {
-		if (astNodeParent instanceof MessageSend m && this.expectedTypesPtr == -1) {
+		if (astNodeParent instanceof MessageSend m
+				&& !(m.receiver instanceof CompletionOnSingleNameReference)
```

Comment:
According to the issue description, you say that checkForVarargExpectedTypes isn't checking that the completion is in the arguments.  Shouldn't this test be more specific to ensure we are in the arguments instead of just not CompletionOnSingleNameReference?

Reply:
This method in only called when completing a single name reference, but I made the solution a bit more general